common/scylla_cloud_snitch

```diff
@@ -0,0 +1,186 @@
+        self._data = re.sub(f'^{key}\s?[:=]\s?[^\n]*$', new_line, self._data, flags=re.MULTILINE)
+        self.__load()
+
+    def commit(self):
+        with open(self._filename, 'w') as f:
+            f.write(self._data)
+
+class CloudSnitch(metaclass=ABCMeta):
+    CASSANDRA_RACKDC_PATH = '/etc/scylla/cassandra-rackdc.properties'
+    SCYLLA_YAML_PATH = '/etc/scylla/scylla.yaml'
+
+    def __init__(self):
+        self.dc = None
+        self.rack = None
+        self.prefer_local = None
+        self.dc_suffix = None
+        self.__properties = JavaPropertiesParser(CloudSnitch.CASSANDRA_RACKDC_PATH)
+        self.__load_properties_file()
+
+    def __load_properties_file(self):
+        if self.__properties.has_option('dc'):
+            self.dc = self.__properties.get('dc')
+        if self.__properties.has_option('rack'):
+            self.rack = self.__properties.get('rack')
+        if self.__properties.has_option('prefer_local'):
+            self.prefer_local = self.__properties.get('prefer_local')
+        if self.__properties.has_option('dc_suffix'):
+            self.dc_suffix = self.__properties.get('dc_suffix')
+
+    def update_properties_file(self):
+        if self.dc is None:
+            raise Exception('dc is not set')
+        self.__properties.set('dc', self.dc)
+
+        if self.rack is None:
+            raise Exception('rack is not set')
+        self.__properties.set('rack', self.rack)
+
+        if self.prefer_local is None:
+            raise Exception('prefer_local is not set')
+        self.__properties.set('prefer_local', self.prefer_local)
+
+        if self.dc_suffix is not None:
+            self.__properties.set('dc_suffix', self.dc_suffix)
+        self.__properties.commit()
+        LOGGER.info('Configured cassandra-rackdc.properties with following values: dc=%s rack=%s prefer_local=%s dc_suffix=%s', self.dc, self.rack, self.prefer_local, self.dc_suffix)
+
+    @classmethod
+    def get_snitch_name_from_yaml(cls):
+        with open(CloudSnitch.SCYLLA_YAML_PATH) as f:
+            scylla_yaml = yaml.safe_load(f)
+        return scylla_yaml['endpoint_snitch']
+
+    @classmethod
+    def get_cloud_snitch(cls):
+        snitch_name = cls.get_snitch_name_from_yaml()
+        try:
+            snitch_cls = globals()[snitch_name]
+        except:
+            LOGGER.warning('Does not supported snitch:%s, exiting', snitch_name)
+            sys.exit(0)
+        LOGGER.info('Starting snitch: %s', snitch_name)
+        return snitch_cls()
+
+    @abstractmethod
+    def start(self):
+        pass
+
+class Ec2Snitch(CloudSnitch):
+    def __init__(self):
+        super().__init__()
+        if not is_ec2():
+            raise Exception('Could not detect EC2 instance')
+        self.__aws_instance = aws_instance()
+
+    def start(self):
+        # Split "us-east-1a" or "asia-1a" into "us-east"/"1a" and "asia"/"1a".
+        splited_zone = self.__aws_instance.availability_zone().rsplit('-', 1)
```

Comment:
this is only the legacy naming support

I think we should add the support for both: 
https://cassandra.apache.org/doc/latest/cassandra/configuration/cass_rackdc_file.html#ec2_naming_scheme

and consider changing the default to match cassandra 4.x